features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListPresenter.kt

```diff
@@ -115,19 +116,47 @@ class DefaultActionListPresenter @AssistedInject constructor(
             isEventPinned = pinnedEventIds.contains(timelineItem.eventId),
         )
 
-        val displayEmojiReactions = usersEventPermissions.canSendReaction &&
-            timelineItem.content.canReact()
-        if (actions.isNotEmpty() || displayEmojiReactions) {
+        val verifiedUserSendFailure = buildVerifiedUserSendFailure(timelineItem)
+        val displayEmojiReactions = usersEventPermissions.canSendReaction && timelineItem.content.canReact()
+
+        if (actions.isNotEmpty() || displayEmojiReactions || verifiedUserSendFailure != ActionListState.VerifiedUserSendFailure.None) {
             target.value = ActionListState.Target.Success(
                 event = timelineItem,
                 displayEmojiReactions = displayEmojiReactions,
+                verifiedUserSendFailure = verifiedUserSendFailure,
                 actions = actions.toImmutableList()
             )
         } else {
             target.value = ActionListState.Target.None
         }
     }
 
+    private suspend fun buildVerifiedUserSendFailure(
+        timelineItem: TimelineItem.Event,
+    ): ActionListState.VerifiedUserSendFailure {
+        return when (val sendState = timelineItem.localSendState) {
+            is LocalEventSendState.Failed.VerifiedUserHasUnsignedDevice -> {
+                val userId = sendState.devices.keys.firstOrNull()
+                if (userId == null) {
+                    ActionListState.VerifiedUserSendFailure.None
+                } else {
+                    val displayName = room.userDisplayName(userId).getOrNull() ?: userId.value
+                    ActionListState.VerifiedUserSendFailure.UnsignedDevice(displayName)
+                }
+            }
+            is LocalEventSendState.Failed.VerifiedUserChangedIdentity -> {
+                val userId = sendState.users.firstOrNull()
+                if (userId == null) {
+                    ActionListState.VerifiedUserSendFailure.None
```

Comment:
Same here, whether we decide to treat this as 'the SDK failed to provide some user but the status is right' or 'the status is wrong since there is no user' I think we should at least log these issues.